src/main/perf/facets/BenchmarkFacets.java

```diff
@@ -80,81 +83,114 @@ public IndexSearcher newSearcher(IndexReader reader, IndexReader previous) {
-            FacetResult result = taxoFacets.getTopChildren(10, taxoFieldName, "TX");
-            long taxoGetAllChildrenEnd = System.nanoTime();
-            double taxoGetAllChildrenTime = (double) (taxoGetAllChildrenEnd - taxoGetAllDimsStart) / BILLION;
-
-            long ssdvGetChildrenStart = System.nanoTime();
-            result = ssdvFacets.getTopChildren(10, ssdvFieldName, "TX");
-            long ssdvGetChildrenEnd = System.nanoTime();
-            double ssdvGetAllChildrenTime = (double) (ssdvGetChildrenEnd - ssdvGetAllDimsStart) / BILLION;
+            System.out.println("Warming up...");
+
+            // Warmup
+            for (int i = 0; i < 5; i++) {
+                FacetsCollector c = new FacetsCollector();
+                s.search(new MatchAllDocsQuery(), c);
+                Facets taxoFacets = new FastTaxonomyFacetCounts(TAXO_FIELD_NAME, taxoReader, config, c);
+                Facets ssdvFacets = new SortedSetDocValuesFacetCounts(state, c);
+                taxoFacets.getAllDims(5);
+                ssdvFacets.getAllDims(5);
+                taxoFacets.getTopChildren(10, TAXO_FIELD_NAME, "TX");
+                ssdvFacets.getTopChildren(10, SSDV_FIELD_NAME, "TX");
+            }
 
-            System.out.println("Time (s) taken to get all dims for taxonomy: " + taxoGetAllChildrenTime);
-            System.out.println("Time (s) taken to get all dims for SSDV: " + ssdvGetAllChildrenTime);
-            System.out.println("Percent difference: " + getPercentDiff(taxoGetAllChildrenTime, ssdvGetAllChildrenTime) + "%");
+            System.out.println("Number of docs: " + s.getIndexReader().numDocs());
+            System.out.println("Number of segments: " + s.getIndexReader().leaves().size());
 
+            System.out.println("Running benchmark...");
+
+            int numIters = 20;
+
+            double totalCreateFastTaxoCountsTime = 0;
+            double totalCreateSSDVCountsTime = 0;
+            double totalTaxoGetAllDimsTime = 0;
+            double totalSSDVGetAllDimsTime = 0;
+            double totalTaxoGetChildrenTime = 0;
+            double totalSSDVGetChildrenTime = 0;
+
+            for (int i = 1; i < numIters + 1; i++) {
+                FacetsCollector c = new FacetsCollector();
+                s.search(new MatchAllDocsQuery(), c);
+
+                long createFastTaxoCountsStart = System.nanoTime();
+                Facets taxoFacets = new FastTaxonomyFacetCounts(TAXO_FIELD_NAME, taxoReader, config, c);
+                long createFastTaxoCountsEnd = System.nanoTime();
+                double createFastTaxoCountsTime = (double)(createFastTaxoCountsEnd - createFastTaxoCountsStart) / MILLION;
+                totalCreateFastTaxoCountsTime += createFastTaxoCountsTime;
+
+                long createSSDVReaderStateStart = System.nanoTime();
+                Facets ssdvFacets = new SortedSetDocValuesFacetCounts(state, c);
+                long createSSDVReaderStateEnd = System.nanoTime();
```

Comment:
Hmm this is not creating the reader state right?  It's actually computing the SSDV facets?